src/cljs/chronograph_web/db.cljs

```diff
@@ -1,5 +1,54 @@
 (ns chronograph-web.db)
 
-(def default-db {:user {;; signin-state can be:
-                        ;; :signed-in, :signed-out or :fetching-profile
-                        :signin-state :fetching-profile}})
+(def default-db {:user {:signin-state :fetching-profile}
+                 :page {:handler :root}})
+
+(def conj-to-set (fnil conj #{}))
+
+(defn add-to-set-in
+  "Adds v to a set in the db at the given path."
+  [db path v]
+  (update-in db
+             path
+             conj-to-set
+             v))
+
+(defn current-page-name
+  [db]
+  (get-in db [:page :handler]))
+
+;; Page state functions
+;; Use these functions to get and set state which is specific to
+;; the current page.
+
+(defn get-in-page-state
+  [db path]
+  (get-in db (concat [:page-state (current-page-name db)] path)))
+
+(defn set-in-page-state
+  [db path value]
+  (assoc-in db (concat [:page-state (current-page-name db)] path) value))
+
+(defn update-in-page-state
+  [db path f & args]
+  (apply update-in
+         db
+         (concat [:page-state (current-page-name db)] path)
+         f
+         args))
+
+;; Error reporting functions
+;; Use these functions to report and retrieve errors which
+;; are specific to the current page.
+
+(defn report-error
+  [db error]
+  (update-in-page-state db [:errors] conj-to-set error))
+
+(defn remove-error
+  [db error]
+  (update-in-page-state db [:errors] disj error))
+
+(defn get-errors
+  [db]
+  (get-in-page-state db [:errors]))
```

Comment:
What are we getting from putting this in the page state instead of top level state?

Reply:
Well I thought that the errors could be page-specific, and each page can decide how to render them. If they're page-specific, it makes sense to put them into page state so as to avoid clashes.

On the other hand, we could have a global set of errors, and a common component across pages that somehow renders these errors at the top.

What did you have in mind?

Reply:
From what it looks like right now, these are still externally visible as global, correct? Even if they are internally by page-state.